src/converters/mod.rs

```diff
@@ -0,0 +1,70 @@
+mod scalars;
+mod traits;
+mod transforms;
+
+use scalars::*;
+use traits::*;
+use transforms::*;
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use std::io::Cursor;
+
+use rerun;
+
+use anyhow::{Error, Result};
+
+pub struct ConverterRegistry {
+    converters: HashMap<String, Arc<dyn Converter>>,
+}
+
+impl ConverterRegistry {
+    pub fn new() -> Self {
+        Self {
+            converters: HashMap::new(),
+        }
+    }
+
+    pub fn register(&mut self, name: &str, converter: Arc<dyn Converter>) {
+        self.converters.insert(name.to_string(), converter);
+    }
+
+    pub fn get(&self, name: &str) -> Option<&Arc<dyn Converter>> {
+        self.converters.get(name)
+    }
+
+    pub fn process(
+        &self,
+        rec: &Arc<rerun::RecordingStream>,
+        ros_type: &str,
+        message: &mut Cursor<Vec<u8>>,
+    ) -> Result<(), Error> {
+        let converter = self.get(ros_type).unwrap();
```

Comment:
I believe this is where we need to look up which conversion is happening from the configuration based on `(topic, frame_id)` in order to then pass that config to the converter, to know where the routing is happening.